src/net/dns.cc

```diff
@@ -1111,9 +1111,9 @@ dns_resolver::impl::do_sendv(ares_socket_t fd, const iovec * vec, int len) {
             switch (e.typ) {
             case type::tcp:
                 if (!e.tcp.out) {
-                    e.tcp.out = e.tcp.socket.output(0);
+                    e.tcp.out = e.tcp.socket.output(0).detach();
                 }
-                f = e.tcp.out->write(std::move(p));
+                f = e.tcp.out->put(std::move(p));
```

Comment:
I see there is no flushing, either before or after. Does output_stream::write(packet) imply a flush? What about data_sink::put(packet)?

Seems wrong to have different overloads of the same function imply or not imply a flush.

Reply:
Since the stream is created with buffer size of 0, calling write(net::packet) into it flushes immediately. And this flush is output_stream::flush(), not data_sink::flush(), it purely does _fd.put(), so this change should be idempotent.

Also, sockets don't implement their data_sink_impl::flush(), I guess the data_sink::flush() was intended to work for file sinks only.